protocols/client/client.go

```diff
@@ -410,25 +419,20 @@ func HybridVSockDialer(sock string, timeout time.Duration) (net.Conn, error) {
 			return nil, err
 		}
 
-		// Receive the packet from the connection without removing it from
-		// the receive queue (MSG_PEEK), ensuring the connection is usable.
-		if uc, ok := conn.(*net.UnixConn); ok {
-			file, err := uc.File()
-			if err != nil {
-				conn.Close()
-				return nil, err
-			}
-			eot := make([]byte, 1)
-			n, _, err := unix.Recvfrom(int(file.Fd()), eot, syscall.MSG_PEEK)
-			file.Close()
-			if err != nil || n == 0 {
-				conn.Close()
-				if err == nil {
-					err = io.EOF
-				}
-				return nil, err
-			}
+		// A trivial handshake is included in the host-initiated vsock connection protocol.
+		// It looks like this:
+		// - [host] CONNECT <port><LF>
+		// - [guest/success] OK <assigned_host_port><LF>
+		reader := bufio.NewReader(conn)
+		response, err := reader.ReadString('\n')
+		if err != nil {
+			conn.Close()
+			return nil, err
+		} else if !strings.Contains(response, "OK") {
+			conn.Close()
+			return nil, errors.New("malformed response code")
```

Comment:
Try removing this line, this way the function will return `conn, nil` and the GRPC code will handle dialing again.